src/ast/mod.rs

```diff
@@ -1401,6 +1401,8 @@ pub enum Statement {
         /// Hive allows you specify whether the table's stored data will be
         /// deleted along with the dropped table
         purge: bool,
+        /// MySQL-specific "TEMPORARY" keyword
+        temporary: Option<bool>,
```

Comment:
I recommend making this `temporary: bool` rather than trying to distinguish between when it was  supported and not (Some/None)

I think this change would make writing the tests easier and it would make the `temporary` flag more consistent with the rest of the codebase